scripts/add_electricity.py

```diff
@@ -385,6 +385,38 @@ def attach_wind_and_solar(
                 efficiency=costs.at[suptech, "efficiency"],
             )
 
+            if tech == "csp":  # add store and link for CSP
+                csp_buses_i = n.madd(
+                    "Bus",
+                    ds.indexes["bus"] + " " + tech,
+                    carrier=tech,
+                    unit="MW",
+                )
```

Comment:
Is there a particular need for assignment? My feeling is that `n.madd()` should work fine to add the the buses to a network.

Reply:
I picked it up from the `extra_components` scripts. It seems to be leading to issues with attaching links to the network.

Reply:
fixed